docfx_yaml/writer.py

```diff
@@ -889,11 +889,15 @@ def visit_reference(self, node):
             if 'http' in node.attributes['refuri']:
                 self.add_text('[{}]({})'.format(node.astext(), node.attributes['refuri']))
             else:
-                # TODO: refactor? so 'hardcoding'
-                # no need '.html#id' ending in yml files
-                pos = node.attributes['refuri'].find('.html')
-                if pos != -1:
-                     node.attributes['refuri'] = node.attributes['refuri'][0: pos]
+                # only use id in class and func refuri if its id exists
+                # otherwise, remove '.html#' in refuri
+                uri_fields = node.attributes['refuri'].split('#')
```

Comment:
uri_fields[1] is bookmark. e.g:
case 0 - [module]#[class-uid]  (using class-uid instead)
case 1 - [module]#module-[module]  (skip)